asneditor.cpp

```diff
@@ -81,3 +90,58 @@ void AsnEditorWidget::unCommentSelection()
 {
     Utils::unCommentSelection(this, m_commentDefinition);
 }
+
+void AsnEditorWidget::contextMenuEvent(QContextMenuEvent *e)
+{
+    QPointer<QMenu> menu(new QMenu(this));
+
+    ActionContainer *mcontext = ActionManager::actionContainer(Constants::M_CONTEXT);
+    QMenu *contextMenu = mcontext->menu();
+
+    foreach (QAction *action, contextMenu->actions()) {
+        menu->addAction(action);
+    }
+
+    appendStandardContextMenuActions(menu);
+
+    menu->exec(e->globalPos());
+    if (!menu)
+        return;
+    delete menu;
```

Comment:
1. `delete nullptr;` is a noop - no need to `if (ptr) delete ptr;`
2. IMHO there is no need to create QMenu via new - it has to exist only in this function, making it on stack will remove the need for `delete`